azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/VirtualMachineScaleSetProperties.java

```diff
@@ -53,8 +51,30 @@ public static VirtualMachineScaleSetProperties create(
       final Boolean overProvision,
       final VirtualMachineScaleSetUpgradePolicy upgradePolicy,
       final VirtualMachineScaleSetVirtualMachineProfile virtualMachineProfile) {
-         return new AutoValue_VirtualMachineScaleSetProperties(
-            singlePlacementGroup, overProvision, upgradePolicy, virtualMachineProfile);
+      return builder().singlePlacementGroup(singlePlacementGroup).overProvision(overProvision).
+         upgradePolicy(upgradePolicy).virtualMachineProfile(virtualMachineProfile).build();
+   }
+
+   public abstract Builder toBuilder();
+
+   private static Builder builder() {
+      return new AutoValue_VirtualMachineScaleSetProperties.Builder();
    }
+
+   @AutoValue.Builder
+   public abstract static class Builder {
+      public abstract Builder singlePlacementGroup(Boolean singlePlacementGroup);
+      public abstract Builder overProvision(Boolean overProvision);
+      public abstract Builder upgradePolicy(VirtualMachineScaleSetUpgradePolicy upgradePolicy);
+      public abstract Builder virtualMachineProfile(VirtualMachineScaleSetVirtualMachineProfile virtualMachineProfile);
+
+      abstract VirtualMachineScaleSetProperties autoBuild();
```

Comment:
This is only needed if you are going to change how the "build" method works. You can remove it and leave just the `public abstract VirtualMachineScaleSetProperties build();`.

Reply:
fixed in last ammend